src/main/java/com/puppycrawl/tools/checkstyle/JavaParser.java

```diff
@@ -146,9 +146,7 @@ public static DetailAST appendHiddenCommentNodes(DetailAST root) {
         DetailAST lastNode = root;
 
         while (curNode != null) {
-            if (isPositionGreater(curNode, lastNode)) {
-                lastNode = curNode;
-            }
+            lastNode = curNode;
```

Comment:
`lastNode` is only used on the very last node. In our case this is either package or class/enum/interface.
Every last node in these cases is always the greatest node, so this is why it can be removed.
Regression showed no differences.